src/instructions/ast.rs

```diff
@@ -260,6 +274,64 @@ impl Register for Value {
         )
     }
 
+    fn clz(&self) -> Value {
+        Value::Op1(ActionOp1::Clz, Rc::new(self.clone()))
+    }
+
+    fn ctz(&self) -> Value {
+        Value::Op1(ActionOp1::Ctz, Rc::new(self.clone()))
+    }
+
+    fn pcnt(&self) -> Value {
+        Value::Op1(ActionOp1::Pcnt, Rc::new(self.clone()))
+    }
+
+    fn rol(&self, rhs: &Value) -> Value {
+        if let (Value::Imm(imm1), Value::Imm(imm2)) = (&self, &rhs) {
```

Comment:
unnecessary & operator
```suggestion
        if let (Value::Imm(imm1), Value::Imm(imm2)) = (self, rhs) {
```